projects/ng-keyboard-shortcuts/src/lib/ng-keyboard-shortcuts.service.ts

```diff
@@ -320,7 +320,11 @@ export class KeyboardShortcutsService implements OnDestroy {
      * @param shortcuts
      */
     private isSequence(shortcuts: string[]): boolean {
-        return !shortcuts.some(shortcut => shortcut.includes("+"));
+        return shortcuts.some(shortcut =>
+            shortcut
+                .split('')
+                .filter((val, index, src) => src[index - 1] !== '+' && val === ' ' && src[index + 1] !== '+')
```

Comment:
think about using reduce here, usually when you need access to previous and next element in iteration, reduce may solve the issue. 
Also feel free to use any of Ramda functions, which I will manually import to this project if needed.
https://ramdajs.com

Reply:
I don't think you can use reduce here, since reduce expects the same type to be returned as is passed in (i.e. a string instead of a boolean in this case). Personally, I think the filter is a nice compromise between being functional, efficient, and being readable without needing a dependency on another third party library.

I still think you would have the same issue if you used ramda, because although you could get a slice using the [aperture ](https://ramdajs.com/docs/#aperture) function you still need to check if there's a blank character that **does not** have a + on either side, which is the tricky bit.

As a side note, i had issues after installing the [types ](https://www.npmjs.com/package/@types/ramda)dependency for ramda which caused the ng build command to get in an infinite loop and fail to compile. Not sure if you have the same issue on your machine or if its just a Windows thing, but it's not as pretty to use ramda without the types dev dependency.

Reply:
Reduce has no limitations with regards to the return type, it takes a list but can return boolean, object, array. Filter can be implemented using reduce.
Regardless, i have played around with the branch and it seems to have an issue where a sequence like this: "command + a b c" will trigger also if you hold command the whole time. Which makes sense given the code changes, however, i am not sure yet that this is the desire behavior. Looking at mousetrap as my main reference for feature parity and usability.
Really appreciate your effort in this pr and this discussion.

Reply:
And i didn't mean to add Ramada as a dependency. if you find a useful function there, we just copy the code for that function to here, it usually has very small footprints, and their functions usually are independent pieces so you can just grab the function from the source code, no need to add Ramda as a dependency
That's what i meant in my original comment, sorry for not being clearer about it

Reply:
I personally think a simple regex match here will suffice, or instead check for the opposite case, which is easier to identify. i.e not a sequence

Reply:
> Regardless, i have played around with the branch and it seems to have an issue where a sequence like this: "command + a b c" will trigger also if you hold command the whole time. Which makes sense given the code changes, however, i am not sure yet that this is the desire behavior. Looking at mousetrap as my main reference for feature parity and usability.

I did notice this behaviour, but to me it's logical that it continues to work whether you hold command down for the following key inputs or not. When i first encountered the side-effect I did wonder if its worth trying to pursue a solution but could not come up with a use case in which this would actually prove to be an issue. I have not used mousetrap but assume it handles this case differently?

Following on from your comment, i did some more testing and found out that Alt  + a ? does not work. I have not had a chance to look into this.

> I personally think a simple regex match here will suffice, or instead check for the opposite case, which is easier to identify. i.e not a sequence

Yes your probably right, I was trying to think of a simpler way of doing it without Regex's, but in this case its got the better of me!
Regex's defiantly have there place and in this scenario it's probably the cleanest approach.

> Really appreciate your effort in this pr and this discussion.

No worries, I'm enjoying it :)

Reply:
Great job :) 
I usually also try to avoid regex, as they are usually "read-only" code. at least for me.
Regarding "command + a b c":
This is a decision that need to be taken, what I am worried is, that making this change, then change the implementation to work "as expected (only trigger if command + a is clicked, released, then ```b``` and ```c``` are clicked."
This behavior does make sense to me, as the user defined a sequence that includes a combination and a sequence. 
I will have to think this through, implementing the described behavior would require more changes and perhaps some rethinking of the code handle it. 

As for the Alt + ? issue
https://github.com/omridevk/ng-keyboard-shortcuts/pull/39/files#diff-7d82ad1a0257ee53472025de4903a112L200

I suspect the removal of ```this.modifiersOn(event)``` from the code is what causing the issue. 

And again thanks a lot for all your time and effort.